wfsim/strax_interface.py

```diff
@@ -160,6 +177,168 @@ def instruction_from_csv(filename):
+        #TODO make config working with nveto...
+        self.context.config['nv_pmt_ce_factor'] = 1.0
+        self.instructions_nveto, self.nveto_channels, self.nveto_timings=read_optical(self.context.config)
+        
+
+    def set_timing(self,):
+        '''Set timing information in such a way to synchronize instructions for the TPC and nVeto'''
+        logging.info("Setting timings")
+        rate = self.context.config['event_rate']
+        start = self.context.config['event_start']
+        stop=self.context.config['event_stop']
+        n_events = stop-start
+
+        timings= (self.context.config['event_start']/rate + \
+                 wfsim.units.s*np.random.random(n_events)*(n_events)/rate).astype(np.int64)
+        timings.sort()
+
+        #For tpc we have multiple instructions per g4id. For nveto we only have one
+        counter=Counter(self.instructions_epix['g4id'])
+        i_per_id = [counter[k] for k in range(start,stop)]
+        timings_tpc=np.repeat(timings,i_per_id)
+        self.instructions_epix['time']+=timings_tpc
+        if self.context.config['neutron_veto']:
+            self.instructions_nveto['time']=timings[self.instructions_nveto['g4id']]
+    
+    def set_configuration(self,):
+        '''Set chunking configuration and feeds instructions to wfsim'''
+        max_events_per_chunk=500#is this hard coding really nessecairy...
+        chunk_size = np.ceil(max_events_per_chunk/self.context.config['event_rate'])
+        nchunk = np.ceil((self.context.config['event_stop']-self.context.config['event_start'])/ \
+                            (chunk_size*self.context.config['event_rate']))
+        self.context.set_config(dict(chunk_size=chunk_size, nchunk=nchunk,))
+        self.context.set_config(dict(wfsim_instructions=self.instructions_epix,))
+        
+        if self.context.config['neutron_veto']:
+            self.context.set_config(dict(wfsim_nveto_instructions=self.instructions_nveto,
+                                         wfsim_nveto_channels=self.nveto_channels,
+                                         wfsim_nveto_timings=self.nveto_timings))
+
+    def set_instructions(self,):
+        '''Gets instructions from epix and neutron veto if needed. 
+        Afterwards sets the correct timings and configuration'''
+        self.instructions_from_epix()
+        if self.context.config['neutron_veto']:
+            self.instructions_from_nveto()
+        self.set_timing()
+        self.set_configuration()
+
+    def run_strax(self,run_id):
+        '''Runs wfsim up to raw records for tpc and if requisted the nveto.
+            All this setting neutron_veto=True/False is needed to get wfsim to do the right thing
+        '''
+        self.context.set_config(dict(neutron_veto=False))
+        self.context.make(run_id,'raw_records')
+
+        #nveto doesn't have proper afterpulses or noise implementation
+        #Cannot use neutron_veto=True/False here
+        if len(self.targets)>1:
```

Comment:
This means in case of nveto stand alone simulations the user has to use a different tool right?

Reply:
Correct. For nveto standalone this is not needed, you'd simply run RawRecordsFromFaxnVeto directly and you're good